tools/src/test/java/org/apache/kafka/trogdor/coordinator/CoordinatorTest.java

```diff
@@ -102,9 +181,13 @@ public void testCreateTask() throws Exception {
                     build()).
                 waitFor(cluster.coordinatorClient());
 
-            // Re-creating a task with the same arguments is not an error.
-            cluster.coordinatorClient().createTask(
-                new CreateTaskRequest("foo", fooSpec));
+            // Re-creating a task with the same arguments gives a RequestConflictException.
+            try {
+                cluster.coordinatorClient().createTask(new CreateTaskRequest("foo", fooSpec));
+                fail("Expected to get an exception when re-creating the same task");
```

Comment:
Re-creating the same task should not cause an exception.  Requests should be idempotent (so that we can just retransmit requests that got lost.)

Reply:
I'm not sure. This is still idempotent in the sense that a task isn't created. I thought it's somewhat confusing to return a `201 CREATED` when none was created. If you think it's alright, I'll change it

Reply:
Idempotent means the same result is returned for the same input.  Not just a similar one.

Reply:
Technically, idempotence is used in multiple contexts and your definition is stricter than usual for HTTP. PUT is meant to be idempotent and the usual behaviour is to return 201 when an item is created and 200 or 204 when it is updated.

"If a new resource is created, the origin server MUST inform the user agent via the 201 (Created) response. If an existing resource is modified, either the 200 (OK) or 204 (No Content) response codes SHOULD be sent to indicate successful completion of the request."

https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html

Reply:
I guess to me, the important thing here is that software that depends on Trogdor should not "see" a request that is retried as different from one which completed successfully the first time.  In order to do that, the return codes need to be the same, not just similar.  This is similar to how, in Kafka, we don't throw an exception or otherwise behave differently if we needed to send a ProduceRequest more than once.